cmd/support-profile.go

```diff
@@ -18,27 +18,171 @@
 package cmd
 
 import (
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+
 	"github.com/minio/cli"
+	"github.com/minio/madmin-go"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio-go/v7/pkg/set"
+	"github.com/minio/pkg/console"
 )
 
-var supportProfileSubcommands = []cli.Command{
-	supportProfileStartCmd,
-	supportProfileStopCmd,
-}
+// profile command flags.
+var (
+	profileFlags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "duration",
+			Usage: "start profiling for the specified duration in seconds",
+			Value: 10,
+		},
+		cli.StringFlag{
+			Name:  "type",
+			Usage: "profiler type, possible values are 'cpu', 'cpuio', 'mem', 'block', 'mutex', 'trace', 'threads' and 'goroutines'",
+			Value: "cpu,mem,block,mutex,threads,goroutines",
+		},
+	}
+)
 
 var supportProfileCmd = cli.Command{
 	Name:            "profile",
 	Usage:           "generate profile data for debugging",
 	Action:          mainSupportProfile,
 	OnUsageError:    onUsageError,
 	Before:          setGlobalsFromContext,
-	Flags:           globalFlags,
-	Subcommands:     supportProfileSubcommands,
+	Flags:           append(profileFlags, globalFlags...),
 	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Fetch CPU profiling only
+     {{.Prompt}} {{.HelpName}} --type cpu myminio/
+
+  2. Fetch CPU, Memory and Block profiling concurrently
+     {{.Prompt}} {{.HelpName}} --type cpu,mem,block myminio/
+
+  3. Fetch CPU, Memory and Block profiling concurrently for 10 minutes
+     {{.Prompt}} {{.HelpName}} --type cpu,mem,block  --duration 600 myminio/	 
+`,
+}
+
+func checkAdminProfileSyntax(ctx *cli.Context) {
+	s := set.CreateStringSet(string(madmin.ProfilerCPU),
+		string(madmin.ProfilerMEM),
+		string(madmin.ProfilerBlock),
+		string(madmin.ProfilerMutex),
+		string(madmin.ProfilerTrace),
+		string(madmin.ProfilerThreads),
+		string(madmin.ProfilerGoroutines),
+		string(madmin.ProfilerCPUIO))
+	// Check if the provided profiler type is known and supported
+	profilers := strings.Split(strings.ToLower(ctx.String("type")), ",")
+	for _, profiler := range profilers {
+		if profiler != "" {
+			if !s.Contains(profiler) {
+				fatalIf(errDummy().Trace(ctx.String("type")),
+					"Profiler type %s unrecognized. Possible values are: %v.", profiler, s)
+			}
+		}
+	}
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 1 {
```

Comment:
```suggestion
	if len(ctx.Args()) != 1 {
```
This would be simpler